### PR TITLE
feat: add degree growth days for growth calculation

### PR DESCRIPTION
**Pull request by wifimug**
- added method of calculating growth per stage based on daily temperature
- slightly refactored how to retrieve Blender crop objects

### CODE REVIEW

File: src/controllers/crop_controller.py [hunk 1]
```diff
@@ -54,33 +54,44 @@ def setup_crops(self):
 
         collection = bpy.data.collections.get(self.collection_name)
         for obj in bpy.context.scene.objects:
+            # if not obj.name in [plants.name for plants in self.all_plants]:
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:57]: Can we remove this commented code?

File: tests/controllers/test_crop_controller.py [hunk 2]
```diff
@@ -50,45 +49,8 @@ def setUp(self):
         self.expected_weed_list = []
         self.expected_material_name = ["stage1", "stage2", "stage3", "stage4", "stage5",
                                        "stage6", "stage7", "stage8", "stage9", "stage10"]
-        self.num_crops_per_stage = {
-            "stage10": 0,
-            "stage8": 0,
-        }
 
 
-    def test_stage_crop_num_correct(self):
```
Comment (DanielSwarup) [inline tests/controllers/test_crop_controller.py:59]: is this test not required anymore?

Comment (wifimug) [reply tests/controllers/test_crop_controller.py:59]: It's not required anymore as each stage is calculated by growth stage instead. The number won't be fixed.

**Review by SeanChao — APPROVED** [review]
LGTM!

**Review by taibohess — APPROVED** [review]

**Pull request merged**